services/ip-whitelist.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+var request = require('superagent');
+var errorMessages = require('../utils/error-messages');
+var logger = require('./logger');
+var ServiceUrlGetter = require('./service-url-getter');
+var ipUtil = require('ip-utils');
+var P = require('bluebird');
+var _ = require('lodash');
+
+var ipWhitelistRules = null;
+var useIpWhitelist = true;
+
+function retrieve(environmentSecret) {
+  var urlService = new ServiceUrlGetter().perform();
+
+  return new P(function (resolve) {
+    request
+      .get(urlService + '/forest/ip-whitelist-rules')
+      .set('forest-secret-key', environmentSecret)
+      .end(function (error, result) {
+        if (result.status === 200 && result.body && result.body.data) {
+          useIpWhitelist = result.body.data.attributes.use_ip_whitelist;
+          ipWhitelistRules = result.body.data.attributes.rules;
+        } else {
+          if (result.status === 0) {
+            logger.error(errorMessages.IP_WHITELIST.SERVER_DOWN);
+          } else if (result.status === 404) {
+            logger.error(errorMessages.IP_WHITELIST.SECRET_NOT_FOUND);
+          } else if (result.status === 422) {
+            logger.error(errorMessages.IP_WHITELIST.SECRET_AND_PROJECTID_INCONSISTENT);
+          } else {
+            logger.error(errorMessages.IP_WHITELIST.UNEXPECTED, error);
+          }
+        }
+        resolve();
+      });
+  });
+}
+
+function isIpValid(ip) {
+  if (useIpWhitelist) {
+    if (ipWhitelistRules === null) {
+      logger.error(errorMessages.IP_WHITELIST.RULES_NOT_LOADED);
+      return false;
+    }
+
+    return _.some(ipWhitelistRules, function (rule) {
+      return ipUtil.isIpMatchesRule(ip, rule);
+    });
+  }
+
+  return true;
+}
+
+module.exports = {
+  retrieve: retrieve,
+  isIpValid: isIpValid
```

Comment:
isIpValid is more an util than a service function to me (there is no business logic)